libraries/AP_AHRS/AP_AHRS_DCM.cpp

```diff
@@ -792,7 +792,7 @@ AP_AHRS_DCM::drift_correction(float deltat)
     // flat, but still allow for yaw correction using the
     // accelerometers at high roll angles as long as we have a GPS
     if (AP_AHRS_DCM::use_compass()) {
-        if (have_gps() && is_equal(gps_gain,1.0f)) {
+        if (have_gps() && is_equal<float>(gps_gain,1.0f)) {
```

Comment:
Actually, what I had in mind is either enforcing a certain function or casting the parameters in AP for all non-float types. In the long run, I wanted to make the behicle code base compatible with a common template library. I think it will succeed the current implementation. Unfortuantely, types are mixed here everywhere and double is not supported everywhere ..